src/ds++/DracoStrings.hh

```diff
@@ -0,0 +1,166 @@
+//----------------------------------*-C++-*----------------------------------//
+/*!
+ * \file   ds++/DracoStrings.hh
+ * \author Kelly G. Thompson <kgt@lanl.gov
+ * \date   Wednesday, Aug 23, 2017, 12:48 pm
+ * \brief  Enscapulates common string manipulations.
+ * \note   Copyright (C) 2017 Los Alamos National Security, LLC.
+ *         All rights reserved. */
+//---------------------------------------------------------------------------//
+
+#ifndef rtt_dsxx_DracoStrings_hh
+#define rtt_dsxx_DracoStrings_hh
+
+#include "ds++/config.h"
+#include <map>
+#include <sstream>
+#include <string>
+#include <vector>
+
+namespace rtt_dsxx {
+
+//----------------------------------------------------------------------------//
+/*!
+ * \brief Convert a number into a string
+ *
+ * \param[in] num The integral or float-point type number that will be converted
+ *                to a string.
+ * \param[in] precision How many digits will be preserved?
+ * \return A string representation of the numeric value
+ *
+ * \sa http://public.research.att.com/~bs/bs_faq2.html
+ */
+template <typename T>
+std::string to_string(T const num, unsigned int const precision = 23) {
+  std::stringstream s;
+  s.precision(precision);
+  s << num;
+  return s.str();
+}
+
+//----------------------------------------------------------------------------//
+/*!
+ * \brief trim whitespace (or other characters) from before and after main text.
+ *
+ * \param[in] str The string that will be processed
+ * \param[in] whitespace A set of characters that will be removed.
+ * \return A new, probably shortened, string without unwanted leading/training
+ *         characters.
+ */
+DLL_PUBLIC_dsxx std::string trim(std::string const &str,
+                                 std::string const &whitespace = " \t");
```

Comment:
I don't see anything specific in our style guide about this, but perhaps it would be useful to describe default values in the doxygen comments?

Reply:
I updated https://rtt.lanl.gov/redmine/projects/draco/wiki/Coding_Standards#section-23 with an example.